staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -76,6 +76,8 @@ type ObjectInterfaces interface {
 	GetObjectDefaulter() runtime.ObjectDefaulter
 	// GetObjectConvertor is the ObjectConvertor appropriate for the requested object.
 	GetObjectConvertor() runtime.ObjectConvertor
+	// GetResourceMapper is the ResourceMapper appropriate for finding equivalent resources and expected kind for the requested object.
+	GetResourceMapper() runtime.ResourceMapper
```

Comment:
The other interface are retrieve-only, this one allows mutation. It makes it hard to reason about it when the ObjectInterfaces are passed around (like to admission). I would prefer to pass a read-only variant to admission instead.

Reply:
split into readonly and writeable interfaces